Define a priority of -50 on the fos_rest.request_body converter

Ener-Getick opened the pull request:
Fix https://github.com/FriendsOfSymfony/FOSRestBundle/issues/1237

Ener-Getick commented:
After all I'm not sure this fixes https://github.com/FriendsOfSymfony/FOSRestBundle/issues/1237 but anyway I think it's better to execute our param converter after the others as it is less restrictive (its ``supports`` function doesn't check anything).

I think we should look at https://github.com/FriendsOfSymfony/FOSRestBundle/issues/1237#issuecomment-160698182 instead for fixing #1237 but this won't be a BC break so we can do it later.

lsmith77 commented:
I would say .. lets merge and see what reports we get during the RC phase

lsmith77 commented:
but it would be good to have a test case for this ..

Ener-Getick commented:
@lsmith77 I fixed #1237 by not throwing exceptions when a parameter is optional.

xabbuh commented on Request/RequestBodyParamConverter.php:
```diff
@@ -95,18 +95,26 @@ public function apply(Request $request, ParamConverter $configuration)
+                throw new BadRequestHttpException($e->getMessage(), $e);
+            }
+        } catch (\Exception $e) {
+            if ($configuration->isOptional()) {
```
What about having this check in each `catch` block or moving it into a private method? I find the nested `try` blocks harder to read.

Ener-Getick replied:
I don't like duplications really much but I agree that it may be the best solution here :-\

xabbuh commented on Tests/Functional/RequestBodyParamConverterTest.php:
```diff
@@ -28,4 +28,13 @@ public function testRequestBodyIsDeserialized()
+    public function testTwigErrorPage()
+    {
+        $client = $this->createClient(['test_case' => 'RequestBodyParamConverter']);
+        $client->request('GET', '/_error/404');
```
Having a test without an assertion looks wrong to me.

Ener-Getick replied:
It is testing that no exceptions are thrown.

xabbuh replied:
I think so. Checking for the proper status code (404 instead of 500) should be enough though.

Ener-Getick commented:
@xabbuh I added a commit with your suggestions.

xabbuh commented:
@Ener-Getick Just to fully understand the changes: What is the underlying in #1237 and how is solved by this PR?

Ener-Getick commented:
@xabbuh The issue in #1237 is that the ``RequestBodyParamConverter`` tries to create a ``Logger`` instance for [this controller](https://github.com/symfony/twig-bundle/blob/master/Controller/ExceptionController.php#L56). 
This behavior is normal because the ``logger`` attribute is not present in the Request attributes as it is optional.

But the issue is that the ``RequestBodyParamConverter`` can't create a ``Logger`` instance from the ``Request`` content so it throws an error.
This PR fixes this by not throwing errors when a parameter is optional.

Ener-Getick commented:
@lsmith77 @xabbuh is this new behavior ok for you ?

lsmith77 commented:
yup .. good for me ... @xabbuh ?

xabbuh commented on Tests/Functional/RequestBodyParamConverterTest.php:
```diff
@@ -28,4 +28,16 @@ public function testRequestBodyIsDeserialized()
+        $client = $this->createClient(['test_case' => 'RequestBodyParamConverter']);
+        $client->request('GET', '/_error/404.txt');
+
+        $this->assertEquals(200, $client->getResponse()->getStatusCode());
```
Why do we have a 200 status code here?

Ener-Getick replied:
Because this page describes an error it is not a page always shown when there is a 404 status code.

xabbuh replied:
Ah, you are right. Thanks. Maybe we should add a comment to avoid confusion for anybody else stumbling over this.

Ener-Getick replied:
Well it seems logic to me. What kind of message do you think about?

xabbuh commented:
I will still need to play with this locally to completely understand the fix, but as this seems to let the original issue go away, we can merge this imo.

xabbuh commented:
By the way, does this affect 1.8 too?

Ener-Getick commented:
No, this is a full bc break not applicable to 1.8. BTW i'll have to add it to the upgrade file.

Ener-Getick commented:
@xabbuh @lsmith77 I updated ``UPGRADING.md`` and I added a comment in ``RequestBodyParamConverterTest`` to explain why the page returns a 200 status.
This PR is ready for me and https://github.com/FriendsOfSymfony/FOSRestBundle/pull/1317 too. Do you see anything else to change before merging them and releasing 1.8/2.0 ?

lyrixx commented:
Hello. Do you have an ETA for the release of this Patch?

Thanks.

Ener-Getick commented:
This is included in 2.0.0-BETA1 but we don't know when we will release it as stable as it depends if we have feedbacks or not.
